src/common/helpers/data-extract/data-extract.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * @typedef  { 'name' | 'file' | 'text' |' radio' | 'address' | 'checkbox' } AnswerType
+ * @typedef  {{ type: AnswerType, value: string, displayText: string }} AnswerData
+ * @typedef  {{ question: string, questionKey: string, answer: AnswerData }} QuestionAnswerData
+ * @typedef  {{ title: string, sectionKey: string, questionAnswers: QuestionAnswerData[]}} SectionData
+ */
+
+/**
+ * @param {string} questionKey
+ * @param {string} sectionKey
+ * @param {SectionData[]} sections
+ * @returns {QuestionAnswerData|undefined}
+ */
+export const getQuestionFromSections = (questionKey, sectionKey, sections) => {
+  const section = Array.isArray(sections)
+    ? sections?.find((section) => section.sectionKey === sectionKey)
+    : {}
+  const question = Array.isArray(section?.questionAnswers)
+    ? section?.questionAnswers?.find(
+        (question) => question.questionKey === questionKey
+      )
+    : undefined
+  return question
+}
+
+/**
+ * @param {object} payload
+ * @returns {SectionData[]|undefined}
+ */
+export const getSectionsFromPayload = (payload) => {
+  if (
+    typeof payload === 'object' &&
+    payload !== null &&
+    'sections' in payload
+  ) {
```

Comment:
And as a result, we can remove this defensive code here?

Reply:
I'll remove all the defensive code and assume schema validation will take care of it in that ticket

Reply:
done